### ISSUE TITLE
my own data set

### ISSUE THREAD

**Issue by oneOfThePeople** [itijyou/ademxapp#29]
hi,
i want to train my own data so i have some question.
1. what is mean --crop-size and --origin-size like if my image in different size what is the  origin-size  and why i need max size also. and crop size why we do it becouse the ram or for the algorithem?
2. my label now are in gray scale 0=bacground 100=myclass 255=ignorlabel there is another format that i need to do for my label and what id_to_label and label_to_id i need to create ? 
thank alot

**Comment by Hustchenqi** [comment]
To my knowledge, id_to_label & label_to_id is only applied to the CityScape dataset. Here is the definition [https://github.com/mcordts/cityscapesScripts/blob/master/cityscapesscripts/helpers/labels.py](https://github.com/mcordts/cityscapesScripts/blob/master/cityscapesscripts/helpers/labels.py). You need do nothing to you label.

The crop size is used because the origin images are randomly re scaled [0.7, 1.3]. This method is used usually to prevent over-fitting.

**Comment by oneOfThePeople** [comment]
thank you.
there is difference between background to ignore label?
how i choose the right crop size to my data ?

**Comment by Hustchenqi** [comment]
If you are doing pixel-level semantic labeling task, there is no difference between background and ignore label.

The crop size depends on your data size and the capacity of your model. 
The author of this project choose 500\*500. 
This [paper](https://hszhao.github.io/projects/pspnet/) choose 713\*713. 
This [paper](https://arxiv.org/abs/1702.08502) choose 544\*544 cropped from 800\*800. 
**All the models run on CityScape dataset**

I think that you can choose your crop size depending on the receptive field of your model. You can also choose a crop size by virtue of experience.

**Comment by oneOfThePeople** [comment]
thank alot  , just for be sure
its mean that my label image should be size hxw with value {0(background) , 1(myclass)} or {255(background) , 1(myclass)}

**Comment by Hustchenqi** [comment]
It looks like that you are doing binary segmentation. {0, 1} is better using **Binary Cross Entropy Loss** function. Wish it to be useful.

**Comment by oneOfThePeople** [comment]
to replace softmax in this - mx.sym.softmax_cross_entropy?
there is somthing else that i need to notice?

**Comment by Hustchenqi** [comment]
No, I mean that Binary Cross Entropy Loss is better if you create your own model. There is no problem using this model directly but changing the class number to 2.

**Comment by oneOfThePeople** [comment]
ok thank you
